Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,301 @@
+
+        Assert(!proposal.Executed, "The proposal is already executed.");
+        Assert(GetVotingDeadline(proposalId) < Block.Number, "Voting is still open for the proposal.");
+        Assert(proposal.RequestedAmount <= Balance, "Insufficient balance.");
+
+        proposal.Executed = true;
+        SetProposal(proposalId, proposal);
+        var result = Transfer(proposal.Recipient, proposal.RequestedAmount);
+
+        Assert(result.Success, "Transfer failed.");
+
+        Log(new ProposalExecutedLog { ProposalId = proposalId, Amount = proposal.RequestedAmount, Recipent = proposal.Recipient });
+
+    }
+
+    public void BlacklistAddress(Address address)
+    {
+        EnsureOwnerOnly();
+
+        if (!IsWhitelisted(address))
+            return;
+
+        SetIsWhitelisted(address, false);
+        WhitelistedCount--;
+    }
+    public void BlacklistAddresses(byte[] addresses)
+    {
+        EnsureOwnerOnly();
+        foreach (var address in Serializer.ToArray<Address>(addresses))
+        {
+            if (!IsWhitelisted(address))
+                continue;
+
+            SetIsWhitelisted(address, false);
+            WhitelistedCount--;
+        }
+    }
+
+    public void WhitelistAddress(Address address)
+    {
+        EnsureOwnerOnly();
+
+        if (IsWhitelisted(address))
+            return;
+
+        SetIsWhitelisted(address, true);
+
+        WhitelistedCount++;
+    }
+
+    public void WhitelistAddresses(byte[] addresses)
+    {
+        EnsureOwnerOnly();
+        foreach (var address in Serializer.ToArray<Address>(addresses))
+        {
+            if (IsWhitelisted(address))
+                continue;
+
+            SetIsWhitelisted(address, true);
+            WhitelistedCount++;
+        }
+    }
+
+    private void EnsureOwnerOnly() => Assert(this.Owner == Message.Sender, "The method is owner only.");
+
+    public void Deposit()
+    {
+        EnsureOwnerOnly();
+        Log(new FundRaisedLog { Sender = Message.Sender, Amount = Message.Value });
+    }
+
+    public override void Receive() => Deposit();
+
+    public void TransferOwnership(Address newOwner)
+    {
+        EnsureOwnerOnly();
+        Owner = newOwner;
+    }
+
+    public void UpdateMinVotingDuration(uint minVotingDuration)
+    {
+        EnsureOwnerOnly();
+
+        Assert(minVotingDuration < MaxVotingDuration, "MinVotingDuration should be lower than MaxVotingDuration.");
+
+        MinVotingDuration = minVotingDuration;
+    }
+
+    public void UpdateMaxVotingDuration(uint maxVotingDuration)
+    {
+        EnsureOwnerOnly();
+        Assert(maxVotingDuration > MinVotingDuration, "MaxVotingDuration should be higher than MinVotingDuration.");
+
+        MaxVotingDuration = maxVotingDuration;
+    }
+
+    public enum Votes : uint
+    {
+        None,
+        No,
+        Yes
+    }
+
+    public struct ProposalAddedLog
+    {
+        public Address Recipent;
```

Comment:
We should index fields we want to search on.

```suggestion
        [Index]
        public Address Recipent;
```